crates/sui-graphql-e2e-tests/tests/call/checkpoint_connection_pagination.move

```diff
@@ -6,99 +6,157 @@
 // Currently implemented only for items ordered in ascending order by `sequenceNumber`.
 
 // Assuming checkpoints 0 through 12
-// first: 4, after: "6" -> checkpoints 7, 8, 9, 10
-// first: 4, after: "6", before: "8" -> checkpoints 7
-// first: 4, before: "6" -> error
-// last: 4, after: "6" -> error
-// last: 4, before: "6" -> checkpoints 2, 3, 4, 5
-// last: 4, before: "6", after: "3" -> error
-// no first or last -> checkpoints 0, 1, 2, 3
+// first: 4, after: 6 -> checkpoints 7, 8, 9, 10
+// first: 4, after: 6, before: 8 -> checkpoints 7
+// first: 4, before: 6 -> checkpoints 0, 1, 2, 3
```

Comment:
This is interesting, my first reaction when reading this was that my mind mapped the result to be 5,4,3,2, because we want 4 checkpoints before checkpoint 6. So I guess first has double semantics: it is a count and from where to start (e.g., from 0 in this case).

Reply:
The cursor specs take some getting used to, and is a little different from how we would interpret it. 
We first found by "before" or "after" if available. So for, 'first 4 before 6', we bound the result set by checkpoints whose cursor are < 6 - before 6. There is no "after" cursor, so we skip that. Then, we apply first, so we select the first 4 elements of the resulting set of [0, 1, 2, 3, 4, 5], which is 0...3. There is no last.

Reply:
Yes, I am starting to get the picture. Would you like to write some docs on the cursor stuff? :D

Reply:
Sure thing, I'm sure there will be a lot of questions around this

Reply:
> Yes, I am starting to get the picture. Would you like to write some docs on the cursor stuff? :D

Haha well all of this stuff is pretty thoroughly documented in our holy text, the [GraphQL Cursor Connection Spec](https://relay.dev/graphql/connections.htm)